### PR TITLE
Added safe-area handling to BottomBannerViewController.

### LINKED ISSUE #1978: NavigationViewController messes up bottom banner height on newer iPhones
Just updated to 0.29 and immediately noticed the bottom banner being off on all full screen (X and newer) iPhones.  I assume this is unintentional as it looks crammed. I had 0.25 before this and it looked great. 

image

Cheers

### PR DESCRIPTION
**Pull request by JThramer**
It's the safety dance!

Fixes #1978. Adds a "padding" view that accounts for the device's bottom safe area, if any.

Test reference:
testbottombannerviewcontroller_iphone_x_portrait_ios_12 1 3x

To-do: 
- [x] Add Snapshot Test
- [x] Changelog entry.

/cc @mapbox/navigation-ios

### CODE REVIEW

**Comment by frederoni** [comment]
An additional view to add some padding seems excessive. Can we constrain the bottom element in the bottom banner with `safeBottomAnchor`?

Also, the NightStyle inherits from the DayStyle, so we need an appropriate color there as well. Can the padding view’s background color be transparent? Much like the UIStackView, a padding view should be a non-rendering view. However, preferably, we shouldn't have a padding view at all.

File: MapboxNavigation/DayStyle.swift [hunk 1]
```diff
@@ -178,6 +179,7 @@ open class NightStyle: DayStyle {
         
         ArrivalTimeLabel.appearance().normalTextColor = #colorLiteral(red: 0.7991961837, green: 0.8232284188, blue: 0.8481693864, alpha: 1)
         BottomBannerView.appearance().backgroundColor = backgroundColor
+        BottomPaddingView.appearance().backgroundColor = #colorLiteral(red: 1, green: 1, blue: 1, alpha: 1)
```
Comment (JThramer) [inline MapboxNavigation/DayStyle.swift]: this should be `backgroundColor`

**Comment by JThramer** [comment]
Posterity: after talking with @frederoni, we decided to make the bottom padding area a `BottomBannerView` type. That way, the padding area can stay consistent with the banner, and we won't have to add extra stuff to the style.

File: Cartfile.private [hunk 2]
```diff
@@ -3,3 +3,4 @@ github "mapbox/MapboxGeocoder.swift" ~> 0.10
 github "Quick/Quick" ~> 1.3
 github "Quick/Nimble" ~> 7.1
 github "CedarBDD/Cedar" ~> 1.0
+github "AndriiDoroshko/SnappyShrimp"
```
Comment (frederoni) [inline Cartfile.private]: nit: Drop `uber/ios-snapshot-test-case` and let 🍤 bring in the correct version.

File: MapboxNavigationTests/BottomBannerSnapshotTests.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    
+    func testBottomBannerViewController() {
+        let host = UIViewController(nibName: nil, bundle: nil)
+        let rootView = host.view!
+        let container = UIView.forAutoLayout()
+        let subject = BottomBannerViewController(nibName: nil, bundle: nil)
+        
+        rootView.addSubview(container)
+        
+        constrain(container, to: rootView)
```
Comment (frederoni) [inline MapboxNavigationTests/BottomBannerSnapshotTests.swift]: nit: inlining this would improve readability. 3 lines vs 9 lines.

File: CHANGELOG.md [hunk 4]
```diff
@@ -36,6 +36,7 @@
 
 ### Other changes
 
+* The `BottomBannerViewController` now accounts for the safe-area inset correctly, if applicable. ([#1982](https://github.com/mapbox/mapbox-navigation-ios/pull/1982))
```
Comment (1ec5) [inline CHANGELOG.md]: This is in the v0.29.0 release notes, which is a couple versions behind. Mind moving it up to a new “master” section?

File: MapboxNavigation/BottomBannerViewController.swift [hunk 5]
```diff
@@ -21,6 +21,16 @@ public protocol BottomBannerViewControllerDelegate: class {
 @objc(MBBottomBannerViewController)
 open class BottomBannerViewController: UIViewController, NavigationComponent {
     
+    /*
+     A padded margin that covers the bottom safe area of the device, if any.
+     */
+    lazy open var bottomPaddingView: BottomBannerView = .forAutoLayout()
```
Comment (1ec5) [inline MapboxNavigation/BottomBannerViewController.swift]: It’s a bit confusing to see a view described as a margin. Can we clarify the comment somehow, perhaps by describing it as a spacer that simulates the margin?

File: MapboxNavigationTests/BottomBannerSnapshotTests.swift [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        
+        embed(parent: host, child: subject, in: container) { (parent, banner) -> [NSLayoutConstraint] in
+            banner.view.translatesAutoresizingMaskIntoConstraints = false
+            return banner.view.constraintsForPinning(to: container)
+        }
+        
+        applyStyling(to: subject)
+        subject.prepareForInterfaceBuilder()
+
+        verify(host, for: Device.iPhoneX.portrait)
```
Comment (1ec5) [inline MapboxNavigationTests/BottomBannerSnapshotTests.swift]: If practical, it would be very nice to also test an older, home button–laden device such as iPhone 8, to make sure there aren’t any issues like zero-height views that confuse Auto Layout.

Comment (frederoni) [reply MapboxNavigationTests/BottomBannerSnapshotTests.swift]: Also, please include the snapshots in the PR description.
